File: Famous Algos/Binary Search/BinarySearch.js

export const binarySearch = (sortedArray,searchTerm)=> {
    let midPointer;
    let startPointer = 0;
    let endPointer = sortedArray.length-1;
    let counter = 1;
    let found = false;
    do{
        midPointer = Math.floor((startPointer + endPointer)/2)
        if (sortedArray[midPointer] === searchTerm){
            found = true;
            break;
        }
        else{
            if (searchTerm<sortedArray[midPointer]){
                endPointer = midPointer - 1
            }
            else{
                startPointer = midPointer + 1
            }
        }
        counter++;
    }while(startPointer<=endPointer)
    
    return [found,counter];
}

File: Famous Algos/Binary Search/BinarySearch.py
import math

def binarySearch(array: [int],targetNum: int):
    startPointer = 0
    endPointer = len(array) -1
    found = False
    counter = 0
    
    while startPointer <= endPointer:
        counter+=1
        midPointer = math.floor((startPointer + endPointer)/2)
        
        if array[midPointer] == targetNum:
            found = True
            break
        else:
            if array[midPointer] > targetNum:
                endPointer = midPointer - 1
            else:
                startPointer = midPointer + 1
        
    return (found,counter)

print(binarySearch([1,2,3,4,5,6,7,8],0))




# O(log n)
        
